app/lib/activitypub/activity/announce.rb

```diff
@@ -35,6 +36,6 @@ def fetch_remote_original_status
   end
 
   def announceable?(status)
-    status.public_visibility? || status.unlisted_visibility?
+    status.account == @account || status.public_visibility? || status.unlisted_visibility?
```

Comment:
Every other part of the source code compares account identifiers instead of instantiated account objects.